thorin/pass/pipelinebuilder.h

```diff
@@ -5,35 +5,90 @@
 
 #include "thorin/pass/optimize.h"
 #include "thorin/pass/pass.h"
+#include "thorin/phase/phase.h"
 
 namespace thorin {
 
 typedef std::function<void(PassMan&)> PassBuilder;
+typedef std::function<void(Pipeline&)> PhaseBuilder;
 typedef std::pair<int, PassBuilder> PrioPassBuilder;
+typedef std::pair<int, PhaseBuilder> PrioPhaseBuilder;
 typedef std::vector<PrioPassBuilder> PassList;
+typedef std::vector<PrioPhaseBuilder> PhaseList;
+// using PassInstanceMap = absl::flat_hash_map<const Def*, Pass*>;
+using PassInstanceMap = std::map<const Def*, Pass*>;
```

Comment:
better use `absl::btree_map<const Def*, Pass*, GIDLt<const Def*>>` - especially the `GIDLt` part is important. `p < q` is UB, if they are not pointing into the same object. While it most of the time does still work, pointer comparisons have the unpleasant effect, that we get different behavior each run. See #143